cpp/ql/src/semmle/code/cpp/rangeanalysis/SimpleRangeAnalysis.qll

```diff
@@ -993,6 +994,18 @@ predicate unanalyzableDefBounds(
   ub = varMaxVal(v)
 }
 
+/**
+ * Holds if in the `branch` branch of `guard`, we know that `v` is not NaN.
+ */
+bindingset[guard, v, branch]
```

Comment:
Is there any difference of communicated intent between `pragma[inline]` and `bindingset[...]`? I did put this annotation in as a response to the typechecker complaining that variables in the body of the predicate were not bound, rather than as a specific thought about performance, if that clarifies anything.

Reply:
That clarifies it. If it's for the sake of the type checker, then `bindingset` is more appropriate.